### PR TITLE
Further fixes to `senior_pointer_first()` and stack initialization

### PR DESCRIPTION
**Pull request by ashelkovnykov**
See [here](https://github.com/urbit/ares/pull/98#issuecomment-1745907804).

### CODE REVIEW

**Comment by ashelkovnykov** [comment]
Need to run now, but realizing that I should add the "pick lower address" logic to the `break` case.

File: rust/ares/src/mem.rs [hunk 1]
```diff
@@ -805,6 +805,11 @@ unsafe fn senior_pointer_first(
     };
 
     loop {
+        if low_pointer.is_null() || high_pointer.is_null() {
+            // we found the top of the stack
+            break (a, b); // both are in the bottom frame, pick arbitrarily
```
Comment (ashelkovnykov) [inline rust/ares/src/mem.rs:810]: Noting for myself that I should copy the logic from lines `819` to `823`.

File: rust/ares/src/mem.rs [hunk 2]
```diff
@@ -70,9 +70,9 @@ impl NockStack {
         let stack_pointer = frame_pointer;
         let alloc_pointer = unsafe { start.add(size) } as *mut u64;
         unsafe {
-            *frame_pointer = ptr::null::<u64>() as u64; // "frame pointer" from "previous" frame
+            *frame_pointer.sub(FRAME) = ptr::null::<u64>() as u64; // "frame pointer" from "previous" frame
```
Comment (eamsden) [inline rust/ares/src/mem.rs:73]: these are all wrong no? They should all be `.sub(CONSTANT + 1)`?

File: rust/ares/src/mem.rs (hunk 1 above)
Comment (eamsden) [inline rust/ares/src/mem.rs:810]: As noted in @ashelkovnykov's comment this should return the lower of the two pointers.

**Review by eamsden — CHANGES_REQUESTED** [review]

File: rust/ares/src/mem.rs (hunk 2 above)
Comment (eamsden) [reply rust/ares/src/mem.rs:73]: Also, the comment above mis-states the test for a west frame, so we should correct it.

Comment (eamsden) [reply rust/ares/src/mem.rs:73]: I think if the above is corrected then the extra null pointer check is unneeded.

**Review by eamsden — CHANGES_REQUESTED** [review]

**Review by eamsden — CHANGES_REQUESTED** [review]

File: rust/ares/src/mem.rs (hunk 2 above)
Comment (eamsden) [reply rust/ares/src/mem.rs:73]: (nevermind this is wrong, you do still need it)

Comment (ashelkovnykov) [reply rust/ares/src/mem.rs:73]: > Also, the comment above mis-states the test for a west frame, so we should correct it.

What's incorrect about this? `frame < alloc` is a valid and correct test for a West frame

**Review by eamsden — APPROVED** [review]

**Pull request merged**